src/Event.php

```diff
@@ -145,6 +168,9 @@ public function setName($name)
      * Set the event target/context
      *
      * @param  null|string|object $target
+     * @template NewTTarget
+     * @psalm-param NewTTarget $target
+     * @psalm-this-out self<TTarget|NewTTarget>
```

Comment:
I think 2 type parameters are to be given here 🤔

Reply:
Yeah I think I got [Psalm's documentation about this](https://psalm.dev/docs/annotating_code/adding_assertions/) wrong the first time around. The union pipe confused me, but I now understand you can just respecify the type. Again, new to Psalm 😅 Will push an update!

Reply:
To be clear, Psalm and PHPStan are pretty much doing the same: they have slight differences in capabilities, but the final types are pretty much identical (and also understood by PHPStan, most of the time) :)